tamr_toolbox/dataset/manage.py

```diff
@@ -0,0 +1,527 @@
+    primary_keys = dataset.spec().to_dict()["keyAttributeNames"]
+
+    # Check input type is correct
+    if attributes and not isinstance(attributes, Iterable):
+        raise TypeError("attributes arg must be an Iterable")
+
+    # Update description and tags
+    dataset_spec = dataset.spec()
+    if description:
+        dataset_spec = dataset_spec.with_description(description)
+        LOGGER.info(f"Updating description for {dataset_name}")
+    if tags:
+        dataset_spec = dataset_spec.with_tags(tags)
+        LOGGER.info(f"Updating tags for {dataset_name}")
+
+    dataset_spec.put()
+
+    if attributes:
+        # Get current dataset attributes
+        target_dataset_attributes = dataset.attributes
+        existing_attributes = []
+        for attr in target_dataset_attributes.stream():
+            existing_attributes.append(attr.name)
+
+        # Update attributes in dataset
+        for attribute_name in attributes:
+            if attribute_name in primary_keys:
+                continue
+            elif attribute_name in existing_attributes:
+                # This attribute already exists, update to new type
+                type_dict = {
+                    attribute_name: (attribute_types or dict()).get(
+                        attribute_name, attribute_type.DEFAULT
+                    )
+                }
+                desc_dict = {
+                    attribute_name: (attribute_descriptions or dict()).get(attribute_name)
+                }
+
+                edit_attributes(
+                    dataset=dataset,
+                    attribute_types=type_dict,
+                    attribute_descriptions=desc_dict,
+                    override_existing_types=override_existing_types,
+                )
+            else:
+                # This attribute does not already exist, create
+                create_attributes(
+                    dataset=dataset,
+                    attributes=[attribute_name],
+                    attribute_types=attribute_types,
+                    attribute_descriptions=attribute_descriptions,
+                )
+
+        # Remove any attributes from dataset that aren't in the new list of attributes
+        for attribute_name in existing_attributes:
+            if attribute_name not in attributes and attribute_name not in primary_keys:
+                delete_attributes(dataset=dataset, attributes=[attribute_name])
+
+    return dataset
+
+
+def create_attributes(
+    *,
+    dataset: Dataset,
+    attributes: Iterable[str],
+    attribute_types: Optional[Dict[str, attribute_type.AttributeType]] = None,
+    attribute_descriptions: Optional[Dict[str, str]] = None,
+) -> Dataset:
+    """
+    Creates new attributes in a dataset. The default attribute type will be ARRAY STRING. To
+    set non-default attribute types, you must define them in the attribute_types dictionary.
+    Any attribute descriptions can be specified in the attribute_descriptions dictionary.
```

Comment:
nit: format and consistent grammer
```suggestion
    """Create new attributes in a dataset

    The default attribute type will be ARRAY STRING. To set non-default attribute types, they must
    be defined in the attribute_types dictionary. Any attribute descriptions can be specified in
    the attribute_descriptions dictionary.
```